modules/3d/test/test_octree.cpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+// I don't know why Octree can't be called directly, I can only test the interface in this way.
+#if 1
+#include "../src/octree.cpp"
+#endif
+
+namespace opencv_test { namespace {
+
+using namespace cv;
+
+class OctreeTest: public::testing::Test
+{
+public:
+    std::vector<Point3f> pointcloud;
+    int pointCloudSize;
+    Point3f restPoint;
+    Octree treeTest;
+
+protected:
+    void SetUp() override
+    {
+        pointCloudSize = 1000;
+        maxDepth = 4;
+        pmin = Point3f(-1.0f, -1.0f, -1.0f);
+        pmax = Point3f(1.0f, 1.0f, 1.0f);
+
+        RNG rng_Point(12345); // set random seed for fixing output 3D point.
+
+        // Generate 3D PointCloud
+        for (int i = 0; i < pointCloudSize; i++)
+        {
+            float _x = rng_Point.uniform(pmin.x, pmax.x);
```

Comment:
@zihaomu, I suggest to use integer pmin & pmax and then inside the loop scale it by a power-of-two, then we can guarantee bit-exactness between platforms:

```
int scale;
Point3i pmin, pmax;
....
scale = 1<<20;
pmin = Point3i(-scale, -scale, -scale);
pmax = Point3i(scale, scale, scale);
...
RNG rng_Point(12345);
```

```
for(int i = 0; i , pointCloudSize; i++)
{
     float _x = (float)rng_Point.uniform(pmin.x, pmax.x)/scale;
     float _y = (float)rng_Point.uniform(pmin.y, pmax.y)/scale;
     float _z = (float)rng_Point.uniform(pmin.z, pmax.z)/scale;
     pointcloud.push_back(Point3f(_x, _y, _z));
}
```